doc/releases/changelog-dev.md

```diff
@@ -49,6 +49,10 @@
 
 <h3>Internal changes ⚙️</h3>
 
+* Catalyst no longer depends on or pins the `scipy` package, instead OpenBLAS is sourced directly
+  from `scipy-openblas32`.
```

Comment:
Worth linking to the PyPI page? (I think this is the only place where the `scipy-openblas32` package is documented)

```suggestion
* Catalyst no longer depends on or pins the `scipy` package, instead OpenBLAS is sourced directly
  from [`scipy-openblas32`](https://pypi.org/project/scipy-openblas32/).
```